golang/json/handler.go

```diff
@@ -117,6 +117,33 @@ func Register(ch *tchannel.Channel, funcs Handlers, onError func(context.Context
 	return nil
 }
 
+func RegisterSub(ch *tchannel.SubChannel, funcs Handlers, onError func(context.Context, error)) error {
```

Comment:
@prashantv - thanks for the suggestion.. removed code duplication now by doing this.. the recent commit should address this..